tests/unit-tests/test_embed.py

```diff
@@ -203,75 +204,15 @@ def test_compute_embed(dev):
 
-def create_lm_graph():
-    bert_model_name = "bert-base-uncased"
-    max_seq_length = 8
-    lm_config = [{"lm_type": "bert",
-                  "model_name": bert_model_name,
-                  "gradient_checkpoint": True,
-                  "node_types": ["n0"]}]
-    with tempfile.TemporaryDirectory() as tmpdirname:
-        # get the test dummy distributed graph
-        g, _ = generate_dummy_dist_graph(tmpdirname)
-
-    feat_size = get_feat_size(g, {'n0' : ['feat']})
-    input_text = ["Hello world!"]
-    tokenizer = AutoTokenizer.from_pretrained(bert_model_name)
-    input_ids, valid_len, attention_mask, _ = \
-        create_tokens(tokenizer=tokenizer,
-                      input_text=input_text,
-                      max_seq_length=max_seq_length,
-                      num_node=g.number_of_nodes('n0'))
-
-    g.nodes['n0'].data[TOKEN_IDX] = input_ids
-    g.nodes['n0'].data[VALID_LEN] = valid_len
-
-    return lm_config, feat_size, input_ids, attention_mask, g
-
-def create_lm_graph2():
-    bert_model_name = "bert-base-uncased"
-    max_seq_length = 8
-    lm_config = [{"lm_type": "bert",
-                  "model_name": bert_model_name,
-                  "gradient_checkpoint": True,
-                  "attention_probs_dropout_prob": 0,
-                  "hidden_dropout_prob":0,
-                  "node_types": ["n0", "n1"]}]
-    with tempfile.TemporaryDirectory() as tmpdirname:
-        # get the test dummy distributed graph
-        g, _ = generate_dummy_dist_graph(tmpdirname)
-
-    feat_size = get_feat_size(g, {'n0' : ['feat']})
-    input_text = ["Hello world!"]
-    tokenizer = AutoTokenizer.from_pretrained(bert_model_name)
-    input_ids0, valid_len0, attention_mask0, _ = \
-        create_tokens(tokenizer=tokenizer,
-                      input_text=input_text,
-                      max_seq_length=max_seq_length,
-                      num_node=g.number_of_nodes('n0'))
-
-    g.nodes['n0'].data[TOKEN_IDX] = input_ids0
-    g.nodes['n0'].data[ATT_MASK_IDX] = valid_len0
-
-    input_text = ["Hello Aamzon!"]
-    input_ids1, valid_len1, attention_mask1, _ = \
-        create_tokens(tokenizer=tokenizer,
-                      input_text=input_text,
-                      max_seq_length=max_seq_length,
-                      num_node=g.number_of_nodes('n1'))
-    g.nodes['n1'].data[TOKEN_IDX] = input_ids1
-    g.nodes['n1'].data[VALID_LEN] = valid_len1
-
-    return lm_config, feat_size, input_ids0, attention_mask0, \
-        input_ids1, attention_mask1, g
-
 def test_lm_infer():
     # initialize the torch distributed environment
     th.distributed.init_process_group(backend='gloo',
                                       init_method='tcp://127.0.0.1:23456',
                                       rank=0,
                                       world_size=1)
-    lm_config, feat_size, input_ids, attention_mask, g = create_lm_graph()
+    with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
is it right that the temporary file will no longer be needed after the graph is constructed?

Reply:
Once we have g and lm_config, we do not need the graph files.